include/pstl_offload/internal/usm_memory_replacement_common.h

```diff
@@ -122,9 +146,15 @@ inline void*
 __realloc_real_pointer(void* __user_ptr, std::size_t __new_size)
 {
     assert(__user_ptr != nullptr);
-    __block_header* __header = static_cast<__block_header*>(__user_ptr) - 1;
+
```

Comment:
Generally, I tried to follow MSDN. `realloc(nullptr, 1024)` is just `malloc()`.

Sadly, I've stared to add `_invalid_parameter_noinfo()` later and may miss some cases. I'll re-read them once more.

Reply:
Aha, I understand.  The assertion for our own purposes, but if `nullptr` is provided to the public API, we should end up as `malloc()`, rather than following this path. If we see `nullptr` here its not a invalid parameter from the user, but rather a bug in our own code.  Thanks for the clarification.